### PR TITLE
Accept valid set samples without restricting sequence length

### PR DESCRIPTION
**Pull request by OswaldHe**

### CODE REVIEW

**Review by ioeddk — APPROVED** [review]

**Review by ioeddk — APPROVED** [review]

**Comment by ioeddk** [comment]
The PR looks good to me. I saw that the logic to compute the Rouge-L score and the mask size in the training loop has changed. Also, the token size filtering of the validation set is taken out. Also I shouldn't take partial datasets when not shuffling, which was a bug, is now fixed in the commit.
